### PR TITLE
Enable pure client suspense in blocking rendering

### PR DESCRIPTION
**Pull request by huozhi**
Do not fallback to loadable component when `reactRoot` is enabled but without `concurrentFeatures`. Thus we can enable pure client suspense:

`fallback` is always rendered on server side, but client side can work with fully functional suspense.

Closes #28117

### CODE REVIEW

File: test/integration/react-18/test/blocking.js [hunk 1]
```diff
@@ -20,8 +20,6 @@ export default (context, render) => {
     const $ = await get$('/suspense/thrown')
     const html = $('body').html()
     expect(html).toContain('loading')
-    expect(
-      JSON.parse($('#__NEXT_DATA__').text()).dynamicIds
-    ).not.toBeUndefined()
+    expect(JSON.parse($('#__NEXT_DATA__').text()).dynamicIds).toBeUndefined()
```
Comment (huozhi) [inline test/integration/react-18/test/blocking.js:23]: the only core change is here, using suspense + react.lazy won't have preload ids like loadable components.

File: test/integration/react-18/test/index.test.js [hunk 2]
```diff
@@ -186,3 +194,8 @@ function runTests(name, mode, fn) {
     fn(context)
   })
 }
+
+function runTests(name, fn) {
```
Comment (huozhi) [inline test/integration/react-18/test/index.test.js:198]: other test changes are just reflected to this simplification

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.7s | -96ms |
| buildDurationCached | 3.5s | 3.5s | ⚠️ +23ms |
| nodeModulesSize | 49.1 MB | 49.1 MB | -109 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.702 | 2.684 | -0.02 |
| / avg req/sec | 925.17 | 931.51 | +6.34 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.456 | 1.518 | ⚠️ +0.06 |
| /error-in-render avg req/sec | 1716.76 | 1646.49 | ⚠️ -70.27 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| 745.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 23.1 kB | 23.1 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 67 kB | 67 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _app-HASH.js gzip | 980 B | 980 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.64 kB | 2.64 kB | -6 B |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 904 B | 904 B | ✓ |
| image-HASH.js gzip | 4.13 kB | 4.13 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 12.9 kB | 12.9 kB | -6 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _buildManifest.js gzip | 491 B | 491 B | ✓ |
| Overall change | 491 B | 491 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| index.html gzip | 540 B | 540 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 532 B | 532 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -10,7 +10,7 @@ self.__BUILD_MANIFEST = {
     "static\u002Fchunks\u002Fpages\u002Fcss-979928a4957344dd4cfa.js"
   ],
   "/dynamic": [
-    "static\u002Fchunks\u002Fpages\u002Fdynamic-cfe35dffa9e5283a3e13.js"
+    "static\u002Fchunks\u002Fpages\u002Fdynamic-383068a57a265c415ff4.js"
   ],
   "/head": ["static\u002Fchunks\u002Fpages\u002Fhead-b93e7fccd3af1c72154e.js"],
   "/hooks": [
```
</details>
<details>
<summary>Diff for <strong>dynamic-HASH.js</strong></summary>

```diff
@@ -137,8 +137,6 @@
               "Disallowed suspense option usage with next/dynamic in blocking mode"
             );
           }
-
-          suspenseOptions.suspense = false;
         }
 
         if (suspenseOptions.suspense) {
```
</details>
</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| buildDuration | 11.1s | 11.3s | ⚠️ +191ms |
| buildDurationCached | 4.6s | 4.6s | ⚠️ +52ms |
| nodeModulesSize | 49.1 MB | 49.1 MB | -109 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.696 | 2.714 | ⚠️ +0.02 |
| / avg req/sec | 927.15 | 921.01 | ⚠️ -6.14 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.461 | 1.537 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1710.93 | 1626.38 | ⚠️ -84.55 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| 16.HASH.js gzip | 186 B | 186 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 14 kB | 14 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 10.6 kB | 10.6 kB | ✓ |
| webpack-HASH.js gzip | 1.19 kB | 1.19 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _app-HASH.js gzip | 965 B | 965 B | ✓ |
| _error-HASH.js gzip | 3.71 kB | 3.71 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.83 kB | 2.83 kB | -5 B |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 2.95 kB | 2.95 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.8 kB | 17.8 kB | -5 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _buildManifest.js gzip | 498 B | 498 B | ✓ |
| Overall change | 498 B | 498 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| index.html gzip | 584 B | 584 B | ✓ |
| link.html gzip | 596 B | 596 B | ✓ |
| withRouter.html gzip | 578 B | 578 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -10,7 +10,7 @@ self.__BUILD_MANIFEST = {
     "static\u002Fchunks\u002Fpages\u002Fcss-0fc40b547d044664b6a0.js"
   ],
   "/dynamic": [
-    "static\u002Fchunks\u002Fpages\u002Fdynamic-03610522cdfd6f22239b.js"
+    "static\u002Fchunks\u002Fpages\u002Fdynamic-bc42be0e3170a6f93209.js"
   ],
   "/head": ["static\u002Fchunks\u002Fpages\u002Fhead-abecfbe6a75f8cfa32c0.js"],
   "/hooks": [
```
</details>
<details>
<summary>Diff for <strong>dynamic-HASH.js</strong></summary>

```diff
@@ -775,8 +775,6 @@ _N_E = (window["webpackJsonp_N_E"] = window["webpackJsonp_N_E"] || []).push([
               "Disallowed suspense option usage with next/dynamic in blocking mode"
             );
           }
-
-          suspenseOptions.suspense = false;
         }
 
         if (suspenseOptions.suspense) {
```
</details>
</details>
Commit: 81d198f02f9277113cf0e46596776e441bf9e960

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| buildDuration | 16s | 16.5s | ⚠️ +454ms |
| buildDurationCached | 3.9s | 4s | ⚠️ +10ms |
| nodeModulesSize | 49.1 MB | 49.1 MB | -111 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.246 | 3.318 | ⚠️ +0.07 |
| / avg req/sec | 770.25 | 753.44 | ⚠️ -16.81 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.946 | 2.022 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1284.5 | 1236.41 | ⚠️ -48.09 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| 745.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 23.1 kB | 23.1 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 67 kB | 67 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _app-HASH.js gzip | 980 B | 980 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.64 kB | 2.64 kB | -6 B |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 904 B | 904 B | ✓ |
| image-HASH.js gzip | 4.13 kB | 4.13 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 12.9 kB | 12.9 kB | -6 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _buildManifest.js gzip | 491 B | 491 B | ✓ |
| Overall change | 491 B | 491 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| index.html gzip | 540 B | 540 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 532 B | 532 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -10,7 +10,7 @@ self.__BUILD_MANIFEST = {
     "static\u002Fchunks\u002Fpages\u002Fcss-979928a4957344dd4cfa.js"
   ],
   "/dynamic": [
-    "static\u002Fchunks\u002Fpages\u002Fdynamic-cfe35dffa9e5283a3e13.js"
+    "static\u002Fchunks\u002Fpages\u002Fdynamic-383068a57a265c415ff4.js"
   ],
   "/head": ["static\u002Fchunks\u002Fpages\u002Fhead-b93e7fccd3af1c72154e.js"],
   "/hooks": [
```
</details>
<details>
<summary>Diff for <strong>dynamic-HASH.js</strong></summary>

```diff
@@ -137,8 +137,6 @@
               "Disallowed suspense option usage with next/dynamic in blocking mode"
             );
           }
-
-          suspenseOptions.suspense = false;
         }
 
         if (suspenseOptions.suspense) {
```
</details>
</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| buildDuration | 13.1s | 14s | ⚠️ +891ms |
| buildDurationCached | 5.6s | 5.4s | -249ms |
| nodeModulesSize | 49.1 MB | 49.1 MB | -111 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.253 | 3.227 | -0.03 |
| / avg req/sec | 768.62 | 774.8 | +6.18 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.097 | 2.02 | -0.08 |
| /error-in-render avg req/sec | 1192.18 | 1237.65 | +45.47 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| 16.HASH.js gzip | 186 B | 186 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 14 kB | 14 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 10.6 kB | 10.6 kB | ✓ |
| webpack-HASH.js gzip | 1.19 kB | 1.19 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _app-HASH.js gzip | 965 B | 965 B | ✓ |
| _error-HASH.js gzip | 3.71 kB | 3.71 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.83 kB | 2.83 kB | -5 B |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 2.95 kB | 2.95 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.8 kB | 17.8 kB | -5 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _buildManifest.js gzip | 498 B | 498 B | ✓ |
| Overall change | 498 B | 498 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| index.html gzip | 584 B | 584 B | ✓ |
| link.html gzip | 596 B | 596 B | ✓ |
| withRouter.html gzip | 578 B | 578 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -10,7 +10,7 @@ self.__BUILD_MANIFEST = {
     "static\u002Fchunks\u002Fpages\u002Fcss-0fc40b547d044664b6a0.js"
   ],
   "/dynamic": [
-    "static\u002Fchunks\u002Fpages\u002Fdynamic-03610522cdfd6f22239b.js"
+    "static\u002Fchunks\u002Fpages\u002Fdynamic-bc42be0e3170a6f93209.js"
   ],
   "/head": ["static\u002Fchunks\u002Fpages\u002Fhead-abecfbe6a75f8cfa32c0.js"],
   "/hooks": [
```
</details>
<details>
<summary>Diff for <strong>dynamic-HASH.js</strong></summary>

```diff
@@ -775,8 +775,6 @@ _N_E = (window["webpackJsonp_N_E"] = window["webpackJsonp_N_E"] || []).push([
               "Disallowed suspense option usage with next/dynamic in blocking mode"
             );
           }
-
-          suspenseOptions.suspense = false;
         }
 
         if (suspenseOptions.suspense) {
```
</details>
</details>
Commit: 6c9fe6990fe33026ebf669c34c93f8e8bf48efaf

**Review by shuding — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12.4s | ⚠️ +100ms |
| buildDurationCached | 3.1s | 3s | -87ms |
| nodeModulesSize | 49.1 MB | 49.1 MB | -111 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.549 | 2.343 | -0.21 |
| / avg req/sec | 980.64 | 1067.06 | +86.42 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.384 | 1.346 | -0.04 |
| /error-in-render avg req/sec | 1805.86 | 1857.08 | +51.22 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| 745.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 23.1 kB | 23.1 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 67 kB | 67 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _app-HASH.js gzip | 980 B | 980 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.64 kB | 2.64 kB | -6 B |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 904 B | 904 B | ✓ |
| image-HASH.js gzip | 4.13 kB | 4.13 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 12.9 kB | 12.9 kB | -6 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _buildManifest.js gzip | 491 B | 491 B | ✓ |
| Overall change | 491 B | 491 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| index.html gzip | 540 B | 540 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 532 B | 532 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -10,7 +10,7 @@ self.__BUILD_MANIFEST = {
     "static\u002Fchunks\u002Fpages\u002Fcss-979928a4957344dd4cfa.js"
   ],
   "/dynamic": [
-    "static\u002Fchunks\u002Fpages\u002Fdynamic-cfe35dffa9e5283a3e13.js"
+    "static\u002Fchunks\u002Fpages\u002Fdynamic-383068a57a265c415ff4.js"
   ],
   "/head": ["static\u002Fchunks\u002Fpages\u002Fhead-b93e7fccd3af1c72154e.js"],
   "/hooks": [
```
</details>
<details>
<summary>Diff for <strong>dynamic-HASH.js</strong></summary>

```diff
@@ -137,8 +137,6 @@
               "Disallowed suspense option usage with next/dynamic in blocking mode"
             );
           }
-
-          suspenseOptions.suspense = false;
         }
 
         if (suspenseOptions.suspense) {
```
</details>
</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| buildDuration | 10s | 9.6s | -406ms |
| buildDurationCached | 4.3s | 4s | -242ms |
| nodeModulesSize | 49.1 MB | 49.1 MB | -111 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.443 | 2.407 | -0.04 |
| / avg req/sec | 1023.5 | 1038.83 | +15.33 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.413 | 1.331 | -0.08 |
| /error-in-render avg req/sec | 1769.23 | 1878.44 | +109.21 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| 16.HASH.js gzip | 186 B | 186 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 14 kB | 14 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 10.6 kB | 10.6 kB | ✓ |
| webpack-HASH.js gzip | 1.19 kB | 1.19 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _app-HASH.js gzip | 965 B | 965 B | ✓ |
| _error-HASH.js gzip | 3.71 kB | 3.71 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.83 kB | 2.83 kB | -5 B |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 2.95 kB | 2.95 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.8 kB | 17.8 kB | -5 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| _buildManifest.js gzip | 498 B | 498 B | ✓ |
| Overall change | 498 B | 498 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | huozhi/next.js rm-dynamic-suspense-fallback | Change |
| - | - | - | - |
| index.html gzip | 584 B | 584 B | ✓ |
| link.html gzip | 596 B | 596 B | ✓ |
| withRouter.html gzip | 578 B | 578 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -10,7 +10,7 @@ self.__BUILD_MANIFEST = {
     "static\u002Fchunks\u002Fpages\u002Fcss-0fc40b547d044664b6a0.js"
   ],
   "/dynamic": [
-    "static\u002Fchunks\u002Fpages\u002Fdynamic-03610522cdfd6f22239b.js"
+    "static\u002Fchunks\u002Fpages\u002Fdynamic-bc42be0e3170a6f93209.js"
   ],
   "/head": ["static\u002Fchunks\u002Fpages\u002Fhead-abecfbe6a75f8cfa32c0.js"],
   "/hooks": [
```
</details>
<details>
<summary>Diff for <strong>dynamic-HASH.js</strong></summary>

```diff
@@ -775,8 +775,6 @@ _N_E = (window["webpackJsonp_N_E"] = window["webpackJsonp_N_E"] || []).push([
               "Disallowed suspense option usage with next/dynamic in blocking mode"
             );
           }
-
-          suspenseOptions.suspense = false;
         }
 
         if (suspenseOptions.suspense) {
```
</details>
</details>
Commit: 9969de9b07d590e8283ddf4da6ecaaae26eb0264

**Review by ijjk — APPROVED** [review]

**Pull request merged**